### DISCUSSION TITLE
Alter schema dynamically at runtime

### DISCUSSION THREAD

**Discussion by JacobWeisenburger** [colinhacks/zod#956] [Q&A] [answered]
Through a series of unfortunate events, it seems that the previous question with this same title was deleted by accident. I am very sorry about this. If you were the person that posted that discussion, could you repost what you had and I will answer it again? Again, very sorry for this inconvenience.

The previous link was: https://github.com/colinhacks/zod/discussions/943. I hope this helps. Maybe there is someway to get the info from your browser cache. My browser didn't have this cached, so I was unable to.

**Comment by sschneider-ihre-pvs** [comment, marked as answer]
Answer was
•if the form is dirty, someField should be z.string().min( 1 ) (not empty string)
•if the form is not dirty, someField should be optional
```js
const isFormDirty = () => { /* implement */ }

const zodSchema = z.object( {
    someField: z.string().optional().refine( input =>
        isFormDirty() ? z.string().min( 1 ).safeParse( input ).success : true
    )
} )
```

**Reply by JacobWeisenburger** [reply]
Thanks so much!

**Comment by sschneider-ihre-pvs** [comment]
in reference to https://github.com/colinhacks/zod/discussions/938 which is still available, but with the difference of having an outside condition for deciding which schema to use.
